src/fft_add.h

```diff
@@ -83,21 +85,24 @@ class Additive : public FourierTransform<T> {
     T m_k, deg0, deg1, deg2;
     T beta_1, inv_beta_1;
     T beta_m, inv_beta_m;
-    vec::Vector<T>* betas = nullptr;
-    vec::Vector<T>* gammas = nullptr;
-    vec::Vector<T>* deltas = nullptr;
-    vec::Vector<T>* beta_m_powers = nullptr;
-    vec::Vector<T>* G = nullptr;
-    vec::Vector<T>* g0 = nullptr;
-    vec::Vector<T>* g1 = nullptr;
-    vec::Vector<T>* u = nullptr;
-    vec::Vector<T>* v = nullptr;
-    vec::Vector<T>* mem = nullptr;
-    Additive<T>* fft_add = nullptr;
+    std::shared_ptr<vec::Vector<T>> betas = nullptr;
```

Comment:
Do we really share the ownership of those vector?

Since they are created by this class, I would expect it to keep the exclusive ownership of the memory (and lending reference to functions that need to use them, but noy sharing the ownership).